### PR TITLE
19334 - Adding Get All Active EFT Organizations support

### PR DESCRIPTION
**Pull request by rodrigo-barraza**
*Issue #:19334*
https://github.com/bcgov/entity/issues/19334

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/services/org.py [hunk 1]
```diff
@@ -708,7 +716,13 @@ def search_orgs(search: OrgSearch, environment):  # pylint: disable=too-many-loc
         }
         include_invitations: bool = False
         search.access_type, is_staff_admin = Org.refine_access_type(search.access_type)
-        if search.statuses and OrgStatus.PENDING_ACTIVATION.value in search.statuses:
+
+        if search.state == EFTShortnameState.UNLINKED.value:
+            eft_payment_accounts = Org.get_eft_payment_accounts()
```
Comment (ochiu) [inline auth-api/src/auth_api/services/org.py:721]: Looks like we are fetching all the eft accounts for every request. After the auth search we should have the ids we need to pass onto pay_api.

File: auth-api/src/auth_api/services/org.py [hunk 2]
```diff
@@ -524,6 +524,14 @@ def get_payment_info(self):
         response = RestService.get(endpoint=f'{pay_url}/accounts/{self._model.id}', token=token, retry_on_failure=True)
         return response.json()
 
+    @staticmethod
+    def get_eft_payment_accounts():
+        """Get the EFT payment accounts."""
+        pay_api_url = current_app.config.get('PAY_API_URL')
+        token = RestService.get_service_account_token()
+        response = RestService.get(endpoint=f'{pay_api_url}/eft-accounts', token=token)
```
[the remark this answers is not in the document]
Comment (seeker25) [reply auth-api/src/auth_api/services/org.py:532]: Alternatively, maybe we could put this in it's own route? not tied into org?

**Review by seeker25 — CHANGES_REQUESTED** [review]
EFT shouldn't be a concept in AUTH, it's a PAY concept

**Comment by ochiu** [comment]
We can probably consider closing this PR once I complete the new simple org search route (I am implementing as part of this ticket).

We should be able to use the simple org search and the eft-shortnames search to get an existing list of LINKED accounts along with a list of the account ids to satisfy #19334 used for the linking modal ticket

**Pull request closed without merging**